lib/rmt/cli/smt_importer.rb

```diff
@@ -0,0 +1,164 @@
+require 'csv'
+require 'ostruct'
+
+
+class SMTImporter
+  attr_accessor :data_dir
+  attr_accessor :no_systems
+
+  class ImportException < StandardError
+  end
+
+  def initialize(data_dir, no_systems)
+    @data_dir = data_dir
+    @no_systems = no_systems
+  end
+
+  def read_csv(file)
+    # set the quote char to something not used to make sure the csv parser is not interfering with the
+    # JSON quoting.
+    CSV.open(File.join(data_dir, file + '.csv'), 'r', { col_sep: "\t", quote_char: "\x00" })
+  end
+
+  def import_repositories
+    read_csv('enabled_repos').each do |row|
+      repo_id, _ = row
+      repo = Repository.find_by(scc_id: repo_id)
+      if repo
+        repo.mirroring_enabled = true
+        repo.save!
+        puts "Enabled mirroring for repository #{repo_id}"
+      else
+        warn "Repository #{repo_id} was not found in RMT database, perhaps you no longer have a valid subscription for it"
+      end
+    end
+  end
+
+  def import_custom_repositories
+    read_csv('enabled_custom_repos').each do |row|
+      product_id, repo_name, repo_url = row
+
+      product = Product.find_by(id: product_id)
+      repo = Repository.find_or_create_by(external_url: repo_url) do |repository|
+        repository.name = repo_name
+        repository.local_path = Repository.make_local_path(repo_url)
+      end
+
+      if product
+        assoc = RepositoriesServicesAssociation.find_by(service: product.service, repository: repo)
+        unless assoc
+          repo.services << product.service
+          puts "Added association between #{repo.name} and product #{product_id}"
+        end
+      else
+        warn "Product #{product_id} not found!"
+        warn "Tried to attach custom repository #{repo.name} to product #{product_id},"
+        warn 'but that product was not found. Attach it to a different product'
+        warn 'by running `rmt-cli repos custom attach`'
+      end
+    end
+  end
+
+  def import_systems
+    read_csv('systems').each do |row|
+      login, password, hostname, registered_at = row
+
+      next unless System.find_by(login: login, password: password).nil?
+
+      # rubocop:disable Rails/TimeZone
+      System.create!(
+        login: login,
+        password: password,
+        hostname: hostname,
+        registered_at: Time.at(registered_at.to_i)
```

Comment:
The timestamp produced by the export script is time in UTC, but `Time.at` interprets it as local time -- so after import you will get a shifted time.

Might not be a big issue, though.

Reply:
Is there a easy solution?

Reply:
`DateTime.strptime("1318996912",'%s')` would convert the timestamp to time in UTC timezone.

Reply:
Looked at this with Ivan, we think this should work:

```
Time.zone = 'UTC' # placing somewhere at the top of the script maybe?
Time.zone.at(1234)
```

Reply:
Apparently it doesn't.
`ActiveRecord::Base.default_timezone = :local` did it for me.

Reply:
OK, we don't need to change anything after all :grimacing: